CMSgov/bcda-app
# BCDA-2596 Maintenance: Remove users table

### Fixes [BCDA-2596](https://jira.cms.gov/browse/BCDA-2596)
Our authentication and authorization no longer has the concept of a user, but jobs and alpha tokens still reference them.  This PR removes all user references.

### Proposed Changes
- Drop `users` table
- Drop `jobs.user_id` column
- Remove all code and fixture references

### Migration Strategy
In `dev` and `test`:
- Deploy code
- `psql -f db/migrations/20200120-drop-users.sql` from deployer's laptop
- Verify success (`users` table should be gone)

In `opensbx` and `prod`:
- Deploy code
- Transfer `db/migrations/20200120-drop-users.sql` to API server in AWS environment
- SSH to server and run `psql -f 20200120-drop-users.sql`
- Verify success (`users` table should be gone)

### Security Implications
- [ ] new software dependencies
- [ ] security controls or supporting software altered
- [ ] new data stored or transmitted
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications

This change should be security-neutral, removing unused database and code artifacts.

### Acceptance Validation
All tests still run ✅ 

### Feedback Requested
Are there any other references?
